### PR TITLE
Journal trivial fixes

### PR DESCRIPTION
**Pull request by daandemeyer**
This is the super trivial stuff from #21183

### CODE REVIEW

File: src/libsystemd/sd-journal/journal-verify.c [hunk 1]
```diff
@@ -824,6 +824,35 @@ static int verify_entry_array(
         return 0;
 }
 
+static int verify_hash_table(
+                Object *o, uint64_t p, uint64_t *n_hash_tables, uint64_t header_offset, uint64_t header_size) {
+
+        if (*n_hash_tables > 1) {
+                error(p, "More than one data hash table: %" PRIu64, *n_hash_tables);
```
Comment (medhefgo) [inline src/libsystemd/sd-journal/journal-verify.c]: It's not really a data hash table if you verify the field hash table... 🤔

Comment (daandemeyer) [reply src/libsystemd/sd-journal/journal-verify.c]: Yikes, fixed

**Review by medhefgo — APPROVED** [review]

File: src/libsystemd/sd-journal/journal-file.c [hunk 2]
```diff
@@ -1972,7 +1972,7 @@ static int journal_file_append_entry_internal(
                 const dual_timestamp *ts,
                 const sd_id128_t *boot_id,
                 uint64_t xor_hash,
-                const EntryItem items[], unsigned n_items,
+                const EntryItem *items, unsigned n_items,
```
Comment (poettering) [inline src/libsystemd/sd-journal/journal-file.c]: why this change btw? we use the syntax sometimes to indicate that we expect more than a single entry. it's entirely equivalent otherwise, so I think it's a good hint.

File: src/libsystemd/sd-journal/journal-def.h [hunk 3]
```diff
@@ -161,23 +161,11 @@ enum {
...
-#elif HAVE_ZSTD
-#  define HEADER_INCOMPATIBLE_SUPPORTED (HEADER_INCOMPATIBLE_COMPRESSED_ZSTD|HEADER_INCOMPATIBLE_KEYED_HASH)
-#else
-#  define HEADER_INCOMPATIBLE_SUPPORTED HEADER_INCOMPATIBLE_KEYED_HASH
-#endif
+#define HEADER_INCOMPATIBLE_SUPPORTED                            \
+        ((HAVE_XZ ? HEADER_INCOMPATIBLE_COMPRESSED_XZ : 0) |     \
+         (HAVE_LZ4 ? HEADER_INCOMPATIBLE_COMPRESSED_LZ4 : 0) |   \
+         (HAVE_ZSTD ? HEADER_INCOMPATIBLE_COMPRESSED_ZSTD : 0) | \
+         HEADER_INCOMPATIBLE_KEYED_HASH)
```
Comment (poettering) [inline src/libsystemd/sd-journal/journal-def.h:168]: some things can be so easy ;-)

File: src/libsystemd/sd-journal/journal-file.c [hunk 4]
```diff
@@ -3206,51 +3207,37 @@ void journal_file_dump(JournalFile *f) {
...
-                        printf("Type: OBJECT_DATA\n");
-                        break;
-
-                case OBJECT_FIELD:
-                        printf("Type: OBJECT_FIELD\n");
-                        break;
+                switch (o->object.type) {
 
                 case OBJECT_ENTRY:
-                        printf("Type: OBJECT_ENTRY seqnum=%"PRIu64" monotonic=%"PRIu64" realtime=%"PRIu64"\n",
```
Comment (poettering) [inline src/libsystemd/sd-journal/journal-file.c:3226]: doesn't matter, but maybe add `assert(s)` here before.

File: src/libsystemd/sd-journal/journal-file.c [hunk 5]
```diff
@@ -3206,51 +3207,37 @@ void journal_file_dump(JournalFile *f) {
...
-                        printf("Type: OBJECT_DATA_HASH_TABLE\n");
-                        break;
-
-                case OBJECT_ENTRY_ARRAY:
-                        printf("Type: OBJECT_ENTRY_ARRAY\n");
-                        break;
-
                 case OBJECT_TAG:
-                        printf("Type: OBJECT_TAG seqnum=%"PRIu64" epoch=%"PRIu64"\n",
+                        printf("Type: %s seqnum=%"PRIu64" epoch=%"PRIu64"\n",
```
Comment (poettering) [inline src/libsystemd/sd-journal/journal-file.c:3233]: ditto

File: src/libsystemd/sd-journal/journal-verify.c [hunk 6]
```diff
@@ -824,6 +824,40 @@ static int verify_entry_array(
         return 0;
 }
 
+static int verify_hash_table(
+                Object *o, uint64_t p, uint64_t *n_hash_tables, uint64_t header_offset, uint64_t header_size) {
```
Comment (poettering) [inline src/libsystemd/sd-journal/journal-verify.c:828]: doesn't matter, but maybe add assert() on the stuff you are about to dereference unconditionally below

File: src/libsystemd/sd-journal/journal-file.c (hunk 2 above)
Comment (daandemeyer) [reply src/libsystemd/sd-journal/journal-file.c]: Ahhh, I thought we always used pointer syntax for arrays, will drop this commit then.

Comment (poettering) [reply src/libsystemd/sd-journal/journal-file.c]: i mean for char arrays we use the pointer syntax. We are not systematic in our use...

But I think in general we should prefer [] syntax for parameters that are array like if the type they are of is not a trivial type or so. but hte lines are blurry.

I mean if you see a * argument it can be one param or more, noone knows, but if you see [] then I think it's clear that it can be multiple.

**Comment by DaanDeMeyer** [comment]
Addressed nits, upgrading green label
